src/containers/Projects/index.js

```diff
@@ -1,41 +1,66 @@
-
+/* @flow */
 
 import React, { Component, PropTypes } from 'react'
 import Helmet from 'react-helmet'
 import { push } from 'react-router-redux'
 import CircularProgress from 'material-ui/CircularProgress'
 import { ErrorMessage, ProjectList } from 'components'
+import GroupList from 'components/GroupList/GroupList'
 import { connect } from 'react-redux'
 import { fetchProjects } from 'ducks/projects'
 import { Sticky, StickyContainer } from 'react-sticky'
 import { Row, Col, Button, ButtonGroup } from 'react-bootstrap'
+import urljoin from 'url-join'
 
 export class Projects extends Component {
   componentDidMount() {
     const { dispatch } = this.props
     this.clickHandler = this.clickHandler.bind(this)
-    // todo: remove this from constructor
     dispatch(fetchProjects())
   }
 
   clickHandler(value) {
-    const { dispatch } = this.props
-    dispatch(push(`/project/${value}`))
+    const { dispatch, pathname } = this.props
+    let path
+    if (value === '') {
+      path = urljoin(pathname.split('/').slice(0, -1))
+    } else {
+      path = urljoin(pathname, value)
+    }
+    dispatch(push(path))
   }
 
   render() {
-    const { isFetching, errors, projects, theme, dispatch } = this.props
+    const { isFetching, errors, tree, theme, dispatch, pathname } = this.props
+    let groups
+    let projects
+    if (pathname === '/projects' || pathname === '/projects/') {
```

Comment:
I would add routes helper here to identify whether the pathname is what we need. Because I would expect someone ask us rename routes to for example `projects` to `repositories` or whatever, so, in that case, we could fix everything in one place.